include/npcomp/Dialect/ATen/ATenOpStatisticsUtils.h

```diff
@@ -0,0 +1,273 @@
+
+  return toReturn;
+}
+
+template<typename T>
+std::map<std::string, uint64_t> getConv2dBackwardStatistics(T op, uint64_t groups) {
+
+  std::map<std::string, uint64_t> toReturn;
+  TensorType dx_out_resultTy = op.getResult(0).getType().template cast<TensorType>();
+  uint64_t dx_out_volume = getTensorVolume(dx_out_resultTy);
+
+  TensorType weightTy = op.getOperand(2).getType().template cast<TensorType>();
+  uint64_t weight_volume = getTensorVolume(weightTy);
+  uint64_t loss_in_depth = weightTy.getShape()[0];
+  uint64_t kernel_width = weightTy.getShape()[2];
+  uint64_t kernel_height = weightTy.getShape()[3];
+
+   uint64_t MACs_per_loss =
+      (loss_in_depth / groups) * kernel_height * kernel_width;
+
+  uint64_t total_MACs = dx_out_volume * MACs_per_loss;
+
+  TensorType ifmTy = op.getOperand(1).getType().template cast<TensorType>();
+  uint64_t ifm_volume = getTensorVolume(ifmTy);
+  auto ifm_shape = ifmTy.getShape();
+
+  uint64_t ifm_bwh = ifm_shape[0] * ifm_shape[2] *
+                     ifm_shape[3]; // Batch * height * width: the depth is in
+                                   // the weight shape already
+  total_MACs += ifm_bwh * weight_volume;
+
+  TensorType dx_inTy = op.getOperand(0).getType().template cast<TensorType>();
+  uint64_t dx_in_volume = getTensorVolume(dx_inTy);
+  toReturn["ops:+"] = dx_in_volume;
+
+  // Reads: Conv_backward reads 3 tensors: the loss in, the activation in and
+  // the transposed weights
+  toReturn["reads"] = dx_in_volume + ifm_volume + weight_volume;
+
+  // Writes: Conv_backward writes 3 tensors: the loss out, gradients for the
+  // weights, and gradients for the biases
+  TensorType biasTy = op.getResult(2).getType().template cast<TensorType>();
+  uint64_t bias_volume = getTensorVolume(biasTy);
+  toReturn["writes"] = dx_out_volume + weight_volume + bias_volume;
+
+  toReturn["ops:MAC"] = total_MACs;
+  toReturn["operand:0:activation_in"] = dx_in_volume;
+  toReturn["operand:1:activation_in"] = ifm_volume;
+  toReturn["operand:2:parameters_in:weight"] = weight_volume;
+
+  toReturn["result:0:grad:dx"] = dx_out_volume;
+  toReturn["result:1:grad:dw"] = weight_volume;
+  toReturn["result:2:grad:db"] = bias_volume;
+
+  return toReturn;
+}
+
+static bool simple_conv2d_model = false;
+
+template <class T>
+uint64_t getConv2dOperandTransferVolume(T *o, unsigned int idx, bool read) {
+
+  if (!read)
+    return 0;
+
+  double vol = getTensorVolume(o->getOperand(idx).getType());
+  if (simple_conv2d_model)
+    return vol;
+
+  TensorType inputTy = o->input().getType().template cast<TensorType>();
+  TensorType weightTy = o->weight().getType().template cast<TensorType>();
+  TensorType resultTy = o->getResult().getType().template cast<TensorType>();
+
+  float filter_width = weightTy.getShape()[2];
+  float filter_height = weightTy.getShape()[3];
+
+  float batch_sw = inputTy.getShape()[0];
+  float ifm_depth_sw = inputTy.getShape()[1];
+  float ih = inputTy.getShape()[2];
```

Comment:
There doesn't seem to be anything in the verifier for the convolution ops that guarantees a particular rank here, so this could easily segfault/abort.

Reply:
Moved to issue.